src/Database/PostgreSQL/PQTypes/Checks.hs

```diff
@@ -819,12 +822,28 @@ checkDBConsistency options domains tablesWithVersions migrations = do
       validateMigrationsToRun migrationsToRun dbTablesWithVersions
       when (not . null $ migrationsToRun) $ do
         logInfo_ "Running migrations..."
-        forM_ migrationsToRun $ \mgr -> do
-          runMigration mgr
-          when (eoCommitAfterEachMigration options) $ do
+        forM_ migrationsToRun $ \mgr -> fix $ \loop -> do
```

Comment:
Shouldn't there be a set amount of attempts? We can ctrl-c the running binary, but if this ever runs automatically, I think it would be better to try, let's say 60x, so that it keeps trying for a minute, and then just bail out.

Reply:
I'd say it's fine - manually you can ctrl+c and if it gets run in Kubernetes, a job is created that can also be cancelled manually (and having to do that should be very rare anyway).